selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/WorkspaceDetailsTest.java

```diff
@@ -0,0 +1,263 @@
+    dashboardWorkspace.checkValue("login", "qaadmin");
+    dashboardWorkspace.clickOnEnvVariableCheckbox("login");
+    dashboardWorkspace.clickOnDeleteBtn();
+    dashboardWorkspace.clickOnDeleteDialogButton();
+    clickOnSaveButton();
+
+    dashboardWorkspace.selectMachine("Environment variables", "db");
+    for (String name : VariablesList) {
+      loader.waitOnClosed();
+      dashboardWorkspace.clickOnDeleteEnvVariableButton(name);
+      dashboardWorkspace.clickOnDeleteDialogButton();
+    }
+    clickOnSaveButton();
+
+    for (int i = 0; i < VariablesList.size(); i++) {
+      loader.waitOnClosed();
+      dashboardWorkspace.clickOnAddEnvVariableButton();
+      dashboardWorkspace.checkAddNewEnvVarialbleDialogIsOpen();
+      dashboardWorkspace.addNewEnvironmentVariable(VariablesList.get(i), ValuesList.get(i));
+      dashboardWorkspace.clickOnAddDialogButton();
+      dashboardWorkspace.checkEnvVariableExists(VariablesList.get(i));
+      dashboardWorkspace.checkValue(VariablesList.get(i), ValuesList.get(i));
+    }
+    clickOnSaveButton();
+  }
+
+  @Test(priority = 1)
+  public void workingWithAgents() {
+    Boolean b;
+    dashboardWorkspace.selectTabInWorspaceMenu(TabNames.AGENTS);
+    dashboardWorkspace.selectMachine("Workspace Agents", "dev-machine");
+    for (String agentName : AgentsList) {
+      dashboardWorkspace.checkAgentExists(agentName);
+    }
+
+    for (int i = 0; i < AgentsList.size(); i++) {
+      b = dashboardWorkspace.getAgentState(AgentsList.get(i));
+      Assert.assertEquals(b, AgentsStateList.get(i));
+      dashboardWorkspace.switchAgentState(AgentsList.get(i));
+    }
+    clickOnSaveButton();
+
+    for (String agentName : AgentsList) {
+      dashboardWorkspace.switchAgentState(agentName);
+    }
+    for (int i = 0; i < AgentsList.size(); i++) {
+      b = dashboardWorkspace.getAgentState(AgentsList.get(i));
+      Assert.assertEquals(b, AgentsStateList.get(i));
+    }
+    clickOnSaveButton();
+  }
+
+  @Test(priority = 2)
+  public void workingWithServers() {
+    dashboardWorkspace.selectTabInWorspaceMenu(TabNames.SERVERS);
+    dashboardWorkspace.selectMachine("Servers", "db");
+    dashboardWorkspace.clickOnAddServerButton();
+    dashboardWorkspace.waitAddServerDialogIsOpen();
+    dashboardWorkspace.enterReference("agen");
+    dashboardWorkspace.enterPort("8080");
+    dashboardWorkspace.enterProtocol("https");
+    dashboardWorkspace.clickOnAddDialogButton();
+    dashboardWorkspace.checkServerExists("agen", "8080");
+    clickOnSaveButton();
+
+    dashboardWorkspace.clickOnEditServerButton("agen");
+    dashboardWorkspace.enterReference("agent");
+    dashboardWorkspace.enterPort("80");
+    dashboardWorkspace.enterProtocol("http");
+    dashboardWorkspace.clickOnUpdateDialogButton();
+    dashboardWorkspace.checkServerExists("agent", "80");
+    dashboardWorkspace.clickOnDeleteServerButton("agent");
+    dashboardWorkspace.clickOnDeleteDialogButton();
+    clickOnSaveButton();
```

Comment:
Why don't we check the result of clicking on save button in the test? 
It looks like the long test divided on several parts which is not the test in fact so as doesn't have separated verification part.